catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -383,6 +411,63 @@ public void init() {
         });
 
     post(
+        "/metacards/deletecheck",
+        (req, res) -> {
+          String shortCircuitStr = req.queryParams("shortcircuit");
+          boolean shortCircuit =
+              isNotEmpty(shortCircuitStr)
+                  ? Boolean.parseBoolean(shortCircuitStr)
+                  : DELETECHECK_SHORTCIRCUIT;
+
+          Set<String> idsToCheck =
+              new HashSet<>(JsonFactory.create().parser().parseList(String.class, req.body()));
+
+          Set<SourceDescriptor> sourceInfos = getSources().getSourceInfo();
+
+          // Produces <number of sources> * <number of partitions> queries
+          List<QueryRequestImpl> queries =
+              Lists.partition(new ArrayList<>(idsToCheck), QUERY_PARTIONING_SIZE)
+                  .stream()
+                  .map(
+                      partitionedIdsToCheck ->
```

Comment:
✏️ This is tough to read. It follows the Spark model so it's going to be a very long `init()` method by nature, but I would strongly suggest breaking out the inner complexities into private methods. It's not that big a deal with the simpler routes, but once you start getting to the reactive streams with nested streams it's a lot to process at one go. I've taken a very quick & ugly swipe at this `deletecheck` route as the worst offender in [this gist](https://gist.github.com/coyotesqrl/6245b085ee340883896252efd41ec02c).

The goal is to separate the flow of the streams from the inner guts to the extent such that the code remains easier to read from a high-level -- what is the overall workflow -- and that the internal complexities can be examined in isolation. Separating at this level decreases the number of cognitive switches a reader needs to make.

Reply:
Done. Will do a larger refactor post-PR.